cmake/modules/soc_v2.cmake

```diff
@@ -28,4 +28,5 @@ if(HWMv2)
   set(SOC_FAMILY ${CONFIG_SOC_FAMILY})
   set(SOC_V2_DIR ${SOC_${SOC_NAME}_DIR})
   set(SOC_FULL_DIR ${SOC_V2_DIR})
+  set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${SOC_V2_DIR}/soc.yml)
```

Comment:
That would be like asking to monitor each and every CMake file, even if it is not part of current build.

Reply:
What?

Reply:
There are 82 soc.yml files in the zephyr tree, whilst this may not seem like much, adding such a check here for every file would mean any time that cmake is used it would need to look up modification dates of those 82 files, on a fragmented HDD where the files could be scattered all over the drive (or even with flash based media on something as slow as an SD card) the time for this is considerable and does not really bring any gains

Reply:
I don't buy the storage media excuse, because CMake already watches thousands of files as part of a normal build.

But I guess what you're really saying is that having stale `build/Kconfig/soc` contents will never be an issue.

Reply:
> What?

What i'm saying is that if you build for an `nrf` SoC, then the file https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/espressif/CMakeLists.txt is never watched by CMake, due to not being included by the build because the soc dir is included in this way: https://github.com/zephyrproject-rtos/zephyr/blob/2396b01b1ea2b0b6a61591155508cf19cd9d367c/soc/CMakeLists.txt#L27

Thus any changes to the espressif SoC CMakeLists.txt doesn't trigger a CMake re-run, and neither should changes to the espressif soc.yml result in a CMake re-run.

In this example, if the `nrf` SoC is removed from the `nordic/soc.yml` (and potentially placed in another file), then a CMake re-run is triggered.

Of course developers can make all sorts of mistakes in files outside CMake watch which may be picked up on a different occasion, like breaking `TF-M` CMakeLists.txt, which is then not discovered until TF-M is included in current build (using Kconfig) or a pristine build including TF-M is invoked. But such mistakes in not intended to be picked up by CMake's own watch system.

Reply:
I think we're talking about two different things here, but I'll let it go.